src/App.jsx

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import classNames from 'classnames';
+import { GoodsList } from './GoodsList';
+import './App.css';
+
+const goodsFromServer = [
+  'Dumplings',
+  'Carrot',
+  'Eggs',
+  'Ice cream',
+  'Apple',
+  'Bread',
+  'Fish',
+  'Honey',
+  'Jam',
+  'Garlic',
+];
+
+class App extends React.Component {
+  state = {
+    goods: [],
+  };
+
+  showGoods = () => {
+    this.setState({
+      goods: [...goodsFromServer],
+    });
+  }
+
+  reverse = () => {
+    this.setState(state => ({
+      goods: [...state.goods].reverse(),
+    }));
+  }
+
+  sortAlphabetically = () => {
+    this.setState(state => ({
+      goods: [...state.goods].sort((a, b) => a.localeCompare(b)),
+    }));
+  }
+
+  sortByLength = () => {
+    this.setState(state => ({
```

Comment:
It would be great to move all the sortings inside the `GoodsList` component.
It's good practice to split the logic. `App` component should only have the button 'start' which will toggle `areGoodsVisible`. And according to the `areGoodsVisible` should render `GoodsList` component. There you should pass initial goods and sort them inside that component (not inside the `App`)